tests/Aql/WaitingExecutionBlockMock.cpp

```diff
@@ -38,16 +38,56 @@ using namespace arangodb::aql;
 using namespace arangodb::tests;
 using namespace arangodb::tests::aql;
 
+namespace {
+static auto blocksToInfos(std::deque<SharedAqlItemBlockPtr> const& blocks) -> ExecutorInfos {
+  auto readInput = make_shared_unordered_set();
+  auto writeOutput = make_shared_unordered_set();
+  std::unordered_set<RegisterId> toClear{};
+  std::unordered_set<RegisterId> toKeep{};
+  RegisterId regs = 1;
+  for (auto const& b : blocks) {
+    if (b != nullptr) {
+      // Find the first non-nullptr block
+      regs = b->getNrRegs();
+
+      break;
+    }
+  }
+  // if non found Sorry blind guess here.
```

Comment:
```suggestion
  // if non found sorry blind guess the number of registers here.
  // This can happen if you insert the data later into this Mock.
  // If you do so this register planning is of
  // for the rime being no test is showing this behavior.
```